[core, lua] Move Spell AoE type and radius calculation to lua

sruon opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Moves all the AoE type and radius modifiers calculation to lua
  - Includes unimplemented Convergence and Theurgic Focus
- Adds accurate String radius bonus for bards
- Introduce several accessors on CSpell to support the changes (getLevel / getRadius)
- Adds a modifier for Utsusemi AoE (Yagyu Darkblade) -- not applied to any item at the moment

## Steps to test these changes

I still need to test a few things on retail first then I'll add xi_test tests.
<!-- Clear and detailed steps to test your changes here -->

sruon committed:
UTSUSEMI_AOE mod to toggle AoE on Utsusemi spells

sruon committed:
Deprecate unused isSpellAoE

sruon committed:
CSpell accessors for radius and level

sruon committed:
Spell AoE type and radius calculations moved to Lua

Xaver-DaRed commented on scripts/globals/magic.lua:
```diff
@@ -148,3 +148,108 @@ function addBonusesAbility(caster, ele, target, dmg, params)
+    end
+
+    return baseType, baseRadius
+end
```
This file is in quarentine. No logic can be added to it. Just removed.

sruon committed:
BLU Convergence Ability script

sruon committed:
GEO Theurgic Focus Ability script

sruon committed:
AoE Spells tests

sruon commented:
- Moved the functions to scripts/globals/combat/magic_aoe.lua
- Added abilities scripts for Theurgic Focus and Convergence 
  - Note that this doesn't meaningfully implement them, it merely applies the restriction associated with the effects
- Added a set of tests
